src/Tizen.Nlp/Tizen.Nlp/NaturalLanguageProcess.cs

```diff
@@ -0,0 +1,372 @@
+            result.Add("token", (string[])msg.GetItem("return_token"));
+            e.RequestId = requestid = int.Parse((string)msg.GetItem("request_id"));
+            if (msg.GetItem("command").Equals("word_tokenize"))
+            {
+                e.Message = result;
+                if (_mapsWordTokenize.ContainsKey(requestid))
+                {
+                    _mapsWordTokenize[requestid]?.Invoke(e);
+                    _mapsWordTokenize.Remove(requestid);
+                }
+            }
+            else if (msg.GetItem("command").Equals("pos_tag"))
+            {
+                result.Add("tag", (string[])msg.GetItem("return_tag"));
+                e.Message = result;
+                if (_mapsPosTag.ContainsKey(requestid))
+                {
+                    _mapsPosTag[requestid]?.Invoke(e);
+                    _mapsPosTag.Remove(requestid);
+                }
+            }
+            else if (msg.GetItem("command").Equals("ne_chunk"))
+            {
+                result.Add("tag", (string[])msg.GetItem("return_tag"));
+                e.Message = result;
+                if (_mapsNamedEntity.ContainsKey(requestid))
+                {
+                    _mapsNamedEntity[requestid]?.Invoke(e);
+                    _mapsNamedEntity.Remove(requestid);
+                }
+            }
+            else if (msg.GetItem("command").Equals("lemmatize"))
+            {
+                e.Message = result;
+                if (_mapsLemmatize.ContainsKey(requestid))
+                {
+                    _mapsLemmatize[requestid]?.Invoke(e);
+                    _mapsLemmatize.Remove(requestid);
+                }
+            }
+            else if (msg.GetItem("command").Equals("langdetect"))
+            {
+                e.Message = result;
+                if (_mapsLangDetect.ContainsKey(requestid))
+                {
+                    _mapsLangDetect[requestid]?.Invoke(e);
+                    _mapsLangDetect.Remove(requestid);
+                }
+            }
+        }
+
+        /// <summary>
+        /// An construct method  to init local env of NLP .
+        /// </summary>
+        /// <since_tizen> 5 </since_tizen>
+        public NaturalLanguageProcess()
+        {
+            _tag = Application.Current.ApplicationInfo.ApplicationId;
+            _noti.Received += ResultReceived;
+        }
+
+
+        /// <summary>
+        /// An async method  to connect remote service.
+        /// </summary>
+        /// <since_tizen> 5 </since_tizen>
+        /// <exception cref="InvalidOperationException">Thrown when the connect is rejected.</exception>
+        public Task Connect()
+        {
+            if (_isConnected)
+            {
+                throw new InvalidOperationException("message already connected");
+            }
+
+            if (_isConnecting)
+            {
+                throw new InvalidOperationException("message is connecting");
```

Comment:
If I were you, I would not throw an exception.
I will return the Task of the previous request.

Because there is no API to get the current connection state. and it is not issue on application returning Task of the previous request.

example code
``` c#
  var nlp = new NaturalLanguageProcess();
  var task = nlp.Connect();
  var task2 = nlp.Connect();
  // task == task2 => // true
```

so, 
``` c#
 if (_isConnected) 
 {
    return Task.CompletedTask;
 }
 if (_isConnecting)
 {
     return _connectionTask;
 }
 ...
 return _connectionTask = tcs.Task;
```